### PR TITLE
feat(platform): bump k8s to 1.21.4-tke.1

### LINKED ISSUE #1508: 【Feature】support kubernetes v1.21
**What would you like to be added**:

Support kubernetes v1.21

### PR DESCRIPTION
**Pull request by TomatoAres**
**What type of PR is this?**

> /kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #1508

### CODE REVIEW

File: build/docker/tools/provider-res/Makefile [hunk 1]
```diff
@@ -15,7 +15,7 @@
 # specific language governing permissions and limitations under the License.
 
 IMAGE = provider-res
-VERSION = v1.20.4-2
+VERSION = v1.21.4-tke
```
Comment (leoryu) [inline build/docker/tools/provider-res/Makefile:18]: should be -1 not -tke

File: pkg/platform/provider/baremetal/cluster/kubeadm.go [hunk 2]
```diff
@@ -126,6 +126,10 @@ func (p *Provider) getInitConfiguration(c *v1.Cluster) *kubeadmv1beta2.InitConfi
 			nodeRegistration.Name = machineIP
 		}
 	}
+
+	// Specify infra(pause) pod version
+	kubeletExtraArgs["pod-infra-container-image"] = p.config.Registry.Prefix + "/pause:3.2"
```
Comment (leoryu) [inline pkg/platform/provider/baremetal/cluster/kubeadm.go:131]: we have method to get images full name, don't hardcode image name

Comment (leoryu) [reply pkg/platform/provider/baremetal/cluster/kubeadm.go:131]: if upgrade from 1.20.4 to 1.21.4, the pause will be which version?

Comment (TomatoAres) [reply pkg/platform/provider/baremetal/cluster/kubeadm.go:131]: pause version is 3.2 in k8s 1.20.x/1.19.x by default with all container runtimes,

and pause version in 3.2 controllered by config in code after k8s 1.21 with all container runtimes,

In a word, in k8s 1.19-1.21  pause version is `3.2` @leoryu

**Pull request merged**